### PR TITLE
Update supported-by information for extension descriptors

### PR DESCRIPTION
**Pull request by nderwin**
In support of #8021, process the supported-by attribute to find a supporter name, then create or update an attribute with the supporter name and the suffix of `-support` to with the status(es) for the extension supporter.  If there is an existing tag ending with `-support`, a `supported-by` tag will be created updated for the supporter(s).

Signed-off-by:Nathan Erwin <nathan.d.erwin@gmail.com>

### CODE REVIEW

File: independent-projects/tools/devtools-common/src/test/resources/datasource-extension.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+metadata:
+  keywords:
+  - "datasource"
+  guide: "https://quarkus.io/guides/datasource"
+  categories:
+  - "data"
+  status: "stable"
+  supported-by: 
+      - "Red Hat"
+  Red Hat-support:
```
Comment (ia3andy) [inline independent-projects/tools/devtools-common/src/test/resources/datasource-extension.yaml:13]: I am not a fan to allow this, a key is supposed to be lower kebabcase

**Comment by ia3andy** [comment]
It looks great. @maxandersen wdyt?

**Comment by gsmet** [comment]
@ia3andy what's the status of this one? FWIW I agree that having things like `Red Hat-support` looks brittle.

If we want a unified support metadata structure, I would probably go with:
```
support:
   - name: Red Hat
     status: stable
   - name: Other company
     status: unstable
```

**Comment by ia3andy** [comment]
> @ia3andy what's the status of this one? FWIW I agree that having things like `Red Hat-support` looks brittle.
> 
> If we want a unified support metadata structure, I would probably go with:
> 
> ```yaml
> support:
>    - name: Red Hat
>      status: stable
>    - name: Other company
>      status: unstable
> ```

yes that works:
```yaml
support:
   - id: redhat 
     name: Red Hat
     status: stable
   - id: other
     name: Other company
     status: unstable
```

**Comment by ia3andy** [comment]
@maxandersen if that structure is ok for you maybe we could re-open this

**Comment by nderwin** [comment]
@ia3andy @maxandersen Based on the conversation above, I've updated the branch (haven't squashed the commits yet) and tests, but had to make a new PR here - #36395.  If that all looks good, I can squash and turn it into a real PR, then go back to https://github.com/quarkusio/code.quarkus.io/pull/1059 to see about displaying the info.

**Pull request closed without merging**